### ISSUE TITLE
Better matching step

### ISSUE THREAD

**Issue by kermitt2** [kermitt2/biblio-glutton#21]
Currently the matching is simply view as a post-validation using reduced metadata (basically author first name, in addition possibly to the title). This is a source of matching error when the title or author names are not present in the bibliographical reference.

To improve matching accuracy, we need a more comprehensive matching step, using more available metadata (provided by the client via the query or via GROBID citation parsing otherwise), while still maintaining robustness given that all these metadata are uncertain.

Example of failing queries: 

- `Tallquist, M.D., Weismann, K.E., Hellstro ̈m, M.,and Soriano, P. (2000). Development 127, 5059–5070.`
->return 10.1038/sj.onc.1203216 while it has no DOI apparently

**Comment by kermitt2** [comment]
Another case of failing query due to insufficient matching step:

`Weeks WB, Bagian JP. Developing a culture of safety in the Veterans Health Administration. Eff Clin Pract. 2007;3:270–276.`

which will match 10.1377/hlthaff.2014.0806 -> same authors list, similar article title (pass ratcliff Obershelp string distance at 0.7), but not the same year and not the same journal+pages !

**Comment by lfoppiano** [comment]
As mentioned in #22, after fixing that bug both this query didn't pass the postValidation

**Comment by kermitt2** [comment]
With version `0.2`:

```
Tallquist, M.D., Weismann, K.E., Hellstro ̈m, M.,and Soriano, P. (2000). Development 127, 5059–5070.
```

is now resolved as  "10.1242/dev.127.23.5059" which indeed matches exactly this reference.

```
Weeks WB, Bagian JP. Developing a culture of safety in the Veterans Health Administration. Eff Clin Pract. 2007;3:270–276.
```

Second case does not pass the validation, which is OK.
